external/googletest-release-1.10.0/CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+# How to become a contributor and submit your own code
```

Comment:
Should these, along with some others,be gitignored?

Reply:
I mean it will be hectic and will need much to go through 257 files in an external library and remove all these comments!

Reply:
@platisd 
Just out of curiosity, 
Is there a better way to include that library into the project (in the same way you can use gradle or maven for instance) or does it need to be "manually" imported? 
Because I don't see the smartCar library being added manually for this project, and as @bassamEldesouki mentioned, it is hectic to go though the whole library and chose what you want from it.

Reply:
@alrapal there are different ways if you don't want to commit those external libraries, however, they are all more difficult than just including it.
One is to have CMake fetch it which is _sometimes_ problematic on Windows. You also need some extra CMake code for that.
Another is to have it as a `git submodule` which means that it would have to be fetched "manually" when cloning the repository.

Reply:
@bassamEldesouki here you go:

https://chromium.googlesource.com/external/github.com/google/googletest/+/refs/heads/master/.gitignore

Reply:
As for the Smartcar library, to use it one has to download it through the Arduino IDE or `arduino-cli`. In other words there's an extra "manual" step involved too.

Reply:
Thanks for these clarifications !

Reply:
@juliavankirk I believe as @platisd said that we shouldn't touch the external library. Also as I said it will be hectic to go through all these files. If we change one thing it might require other implementation and other steps will be involved and I don't know if we have enough time for that. One more important point is that this course is not about CMAKE and what you're asking is involving CMAKE deep understanding. I think this is not required.

Reply:
Remember, the point of this is just to make some unit tests for our code :)

Reply:
I thought we covered some important concerns about the unit testing today in the meeting for around 40 min. On another hand, I agree with the point that a deep understanding of the good libraries, licenses, and general coverage of unit testing is going to take place I believe in the next semester as a whole course. But as I said earlier in the meeting we are still happy to answer your further concerns as much as we can. 

Since we have some unit test now, it is also good clone the repository and play around with it :)